routing/pathfind.go

```diff
@@ -1237,18 +1241,81 @@ func findBlindedPaths(g Graph, target route.Vertex,
 		return features.HasFeature(lnwire.RouteBlindingOptional), nil
 	}
 
+	// Initialize the map to track visited vertices
+	alreadyVisited := map[route.Vertex]bool{}
+	partiallySpecifiedPath := make([]blindedHop, 0)
+
+	// Sanity checking the partially specified blinded path.
+	// Check if the order makes sense (ie, the chain is valid).
+	// Check if the all the channels exist.
+	// Check if all the nodes involved are advertising route blinding.
+	node := target
+	for _, chanID := range restrictions.incomingChainedChannels {
+		// iterate over the node's channels in search for paths to the
+		// currentTarget
+		err := g.ForEachNodeChannel(node,
+			func(channel *graphdb.DirectedChannel) error {
+				if channel.ChannelID == chanID {
+					_, err := supportsRouteBlinding(
+						channel.OtherNode)
+					if err != nil {
+						return err
+					}
+					// Add this node to the visited set.
+					alreadyVisited[node] = true
+					hop := blindedHop{
+						vertex:       channel.OtherNode,
+						channelID:    channel.ChannelID,
+						edgeCapacity: channel.Capacity,
+					}
+					partiallySpecifiedPath =
+						append(partiallySpecifiedPath,
+							hop)
+					node = channel.OtherNode
+				}
+
+				return nil
+			},
+		)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// If the order does not make sense (ie, the partialled specified chain
+	// is not valid) or if at least one channel does not exist.
+	if len(restrictions.incomingChainedChannels) > 0 {
+		if len(restrictions.incomingChainedChannels) !=
+			len(partiallySpecifiedPath) {
+
+			return nil, fmt.Errorf("specified path %v not found",
+				restrictions.incomingChainedChannels)
+		}
+	}
+
 	// This function will have some recursion. We will spin out from the
 	// target node & append edges to the paths until we reach various exit
 	// conditions such as: The maxHops number being reached or reaching
 	// a node that doesn't have any other edges - in that final case, the
 	// whole path should be ignored.
-	paths, _, err := processNodeForBlindedPath(
-		g, target, supportsRouteBlinding, nil, restrictions,
-	)
+	paths, _, err := processNodeForBlindedPath(g, node,
```

Comment:
Q: So we check the length of the `incomingChainedChannels ` and the `restrictions.maxNumHops` beforehand ?